### PR TITLE
Refactor the onError utils method to use the correct translations

### PR DESCRIPTION
**Pull request by FrankeLukas**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I documented the TypeScript code using JSDoc style.

### Motivation and Context

We have a lot of "translation not found" errors on sentry with translation keys which are actually http error messages.

### Description

We display the correct translation for those, and the error message directly if we don't have a translation. We use the shared function for that to reduce duplication.

### Steps for Testing

I'd suggest verifying the changes of this PR by looking at the actual code changes. It should be relatively verifiable that functionality wise nothing is broken. But you can also test if error messages are still displayed, like:
grafik

You can provoke errors by manually enabling disabled buttons only browser side, by deleting `disabled=""` from the element in the DOM explorer:
grafik
->
grafik

That would allow you to e.g. submit a modelling exercise without (max) scores set which causes the server to respond with an error.

### CODE REVIEW

**Review by FuchsDominik — COMMENTED** [review]
Code changes would look good, I thinks it's great because this way we also have consistency in error handling, instead of every component having its own onError method 👍  If the builds passes and you fix the merge conflicts i can approve

**Comment by FrankeLukas** [comment]
Resolved the merge conflicts.

File: src/test/javascript/spec/component/learning-goals/create-learning-goal.component.spec.ts [hunk 1]
```diff
@@ -80,7 +80,8 @@ describe('CreateLearningGoal', () => {
         createLearningGoalComponentFixture.detectChanges();
```
Comment (ivanchimeno) [inline src/test/javascript/spec/component/learning-goals/create-learning-goal.component.spec.ts:80]: You missed a codacy issue here where you need to remove `fakeAsnyc`

Comment (FrankeLukas) [reply src/test/javascript/spec/component/learning-goals/create-learning-goal.component.spec.ts:80]: Thanks, fixed.

File: src/main/webapp/app/course/dashboards/instructor-course-dashboard/instructor-course-dashboard.component.ts [hunk 2]
```diff
@@ -71,8 +72,8 @@ export class InstructorCourseDashboardComponent implements OnInit {
             tap((course: Course) => {
                 this.course = Course.from(course);
             }),
-            catchError((response: HttpErrorResponse) => {
-                this.onError(response.message);
+            catchError((error: HttpErrorResponse) => {
+                onError(this.jhiAlertService, error);
                 return of(null);
             }),
         );
```
Comment (MDK5) [inline src/main/webapp/app/course/dashboards/instructor-course-dashboard/instructor-course-dashboard.component.ts:79]: Further down this file in line 90 theres another onError

File: src/main/webapp/app/exam/manage/test-runs/test-run-management.component.ts [hunk 3]
```diff
@@ -49,13 +50,13 @@ export class TestRunManagementComponent implements OnInit {
                 this.course = this.exam.course!;
                 this.course.isAtLeastInstructor = this.accountService.isAtLeastInstructorInCourse(this.course);
                 this.examManagementService.findAllTestRunsForExam(this.course.id!, this.exam.id!).subscribe(
-                    (res) => {
+                    (res: HttpResponse<StudentExam[]>) => {
                         this.testRuns = res.body!;
                     },
-                    (err) => this.onError(err),
+                    (err: HttpErrorResponse) => onError(this.jhiAlertService, err),
```
Comment (MDK5) [inline src/main/webapp/app/exam/manage/test-runs/test-run-management.component.ts:56]: unifiying err and error would also be a nice change, especially here with both only 3 lines apart

File: src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment.component.ts [hunk 4]
```diff
@@ -353,7 +354,7 @@ export class FileUploadAssessmentComponent implements OnInit, OnDestroy {
                 this.complaint = res.body;
```
Comment (MDK5) [inline src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment.component.ts:354]: A few lines up in 316 theres another error

File: src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts [hunk 5]
```diff
@@ -246,7 +247,7 @@ export class QuizExerciseDetailComponent implements OnInit, OnChanges, Component
                 (res: HttpResponse<string[]>) => {
                     this.existingCategories = this.exerciseService.convertExerciseCategoriesAsStringFromServer(res.body!);
                 },
-                (res: HttpErrorResponse) => this.onError(res),
+                (res: HttpErrorResponse) => onError(this.jhiAlertService, res),
```
Comment (MDK5) [inline src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts:250]: now the error is called res

**Review by MDK5 — CHANGES_REQUESTED** [review]
Reviewed code, changes look good to me, theres only 2 problems I found:
1. In several files the private onError method is not deleted, I suspect in every such file the refactoring is not complete yet (I stopped marking them after 2 files).
2. It would be great if you could also unify the naming of the local error variable, currenty its either `error`, `err`, or `res`; This might be out of scope though.

**Comment by FrankeLukas** [comment]
> In several files the private onError method is not deleted, I suspect in every such file the refactoring is not complete yet (I stopped marking them after 2 files).

Not all service methods return HttpErrorResponses, but sometimes strings for example. Therefore I kept those methods in several places. (Changing all the service methods is Imho out of scope for this PR.)

File: src/main/webapp/app/course/dashboards/instructor-course-dashboard/instructor-course-dashboard.component.ts (hunk 2 above)
Comment (FrankeLukas) [reply src/main/webapp/app/course/dashboards/instructor-course-dashboard/instructor-course-dashboard.component.ts:79]: That method uses a string.

File: src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment.component.ts (hunk 4 above)
Comment (FrankeLukas) [reply src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment.component.ts:354]: That method uses a more specific custom error message, so I don't want to change it.

File: src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts (hunk 5 above)
Comment (FrankeLukas) [reply src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts:250]: Fixed.

File: src/main/webapp/app/exam/manage/test-runs/test-run-management.component.ts (hunk 3 above)
Comment (FrankeLukas) [reply src/main/webapp/app/exam/manage/test-runs/test-run-management.component.ts:56]: Done.

**Review by MDK5 — APPROVED** [review]
Code changes look good to me

**Pull request merged**